### ISSUE TITLE
Handle retries on incoming webhook

### ISSUE THREAD

**Issue by skelz0r** [MicrosoftDocs/msteams-docs#1571]
I experimented that when Microsoft Teams API tries to send an incoming payload to my application it only tries 2 times if it's not a 2XX http status response.

How can I retrieve missing payloads ? Some bot API (like Facebook Messenger API ([source](https://developers.facebook.com/docs/messenger-platform/webhook#response))) retries regularly (until a certain limit).

I saw an another system on DirectLine API with an endpoint to retrieve activities thanks to a watermark parameter ([source](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-direct-line-3-0-receive-activities?view=azure-bot-service-4.0#http-get)) : is it available for Microsoft Teams API ?

Regards.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9da6a6eb-8756-fffd-ecc3-07cde897e709
* Version Independent ID: 72686dc8-d677-9b8d-2405-7684d537a100
* Content: [Handle bot events - Teams](https://docs.microsoft.com/en-us/microsoftteams/platform/resources/bot-v3/bots-notifications#feedback)
* Content Source: [msteams-platform/resources/bot-v3/bots-notifications.md](https://github.com/MicrosoftDocs/msteams-docs/blob/master/msteams-platform/resources/bot-v3/bots-notifications.md)
* Product: **msteams**
* GitHub Login: @laujan
* Microsoft Alias: **lajanuar**

**Comment by Trinetra-MSFT** [comment]
@skelz0r ,  Teams tracks requests via a fixed-rate window and incremental counter measured in seconds. If too many requests are made, the client connection will be throttled until the window refreshes. You can implement retry logic in your code and save the logs. 
Please take a look at [rate limiting for connector](https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using#rate-limiting-for-connectors).

**Comment by skelz0r** [comment]
Thanks for your answer @Trinetra-MSFT 

I'm talking about webhooks from Microsoft Teams API to my application, not from my application to Microsoft Teams API : it makes no sense to speak about rate limit from Microsoft Teams API, which I can't control in my case.

**Comment by skelz0r** [comment]
Hi there,

Any news ? If there's no way to retrieve latest missing payloads, what do you recommend to handle apps restart/bugs ?

**Comment by Trinetra-MSFT** [comment]
@skelz0r , When you are making request to your service through webhook, could you please put it in
```
try
{
//your webhook post request to external service
catch(Exception ex){
//Your code to handle logs failure or retries
} 
```
block and see what error code you're receiving when request fails or you can logs the response and send it to me?

**Comment by skelz0r** [comment]
I think there's a misunderstanding.

When a user sends a message to my bot, Microsoft Teams API sends me a payload through an endpoint which I specified when I created my bot.

Microsoft Teams sends payload to MY endpoint, which is a part that I managed. 

If my endpoint doesn't respond a 2XX (success response), Microsoft Teams doesn't retry to send the payload, which is a huge problem : my webapp can be down for multiple reasons, and if I can't retrieve previously payloads sent by Microsoft Teams I definitely lost activities.
